bublik/core/run/stats.py

```diff
@@ -201,14 +201,15 @@ def add_comments(node, tests_comments):
     }
     The order of comments is determined by the serial value.
     '''
-    node['comments'] = tests_comments.get(node['result_id'], [])
-    node['comments'] = [
-        json.loads(comment) if isinstance(comment, str) else comment
-        for comment in node['comments']
-    ]
-    node['comments'] = sorted(node['comments'], key=lambda x: x['serial'])
-    for child in node['children']:
-        add_comments(child, tests_comments)
+    if node:
```

Comment:
Shouldn't it fail somehow if comments are not actually added?

Reply:
The algorithm recursively traverses the test tree and adds comments to nodes if they are found in the list of comments. If a node does not exist, it cannot have comments or descendants, so the algorithm automatically skips to the next node instead of processing the current subtree.